src/software/simulated_tests/kicker_double_touch.py

```diff
@@ -0,0 +1,66 @@
+import software.python_bindings as tbots_cpp
+from proto.import_all_protos import *
+
+from software.simulated_tests.validation import (
+    Validation,
+    create_validation_geometry,
+    create_validation_types,
+)
+
+
+class KickerDoubleTouch(Validation):
+
+    """Checks if the robot kicking the ball touches the ball again before another robot touches it."""
+
+    def __init__(self):
+        self.kicker_robot = None
+        self.kick_position = None
+        self.other_robot = None
+
+    def get_validation_status(self, world) -> ValidationStatus:
+        """Checks if the robot kicking the ball touches the ball again before another robot touches it
+
+        :param world: The world msg to validate
+        :returns: FAILING when the robot kicking the ball touches the ball again before another robot touches it
+                  PASSING when the robot kicking the ball does not touch the ball again before another robot touches it
+        """
+        ball_position = tbots_cpp.createPoint(world.ball.current_state.global_position)
+        # quick passing check
+        if self.other_robot is not None:
+            return ValidationStatus.PASSING
+        # main check loop
+        for robot in world.friendly_team.team_robots:
+            if not tbots_cpp.Robot(robot).isNearDribbler(ball_position, 0.01):
+                continue
+            elif tbots_cpp.Robot(robot).isNearDribbler(ball_position, 0.01):
+                if self.kicker_robot is None:
+                    self.kicker_robot = robot
+                    self.kick_position = ball_position
+                elif self.kicker_robot == robot and self.other_robot is None:
+                    return ValidationStatus.FAILING
```

Comment:
return FAILING only if the ball has moved > 0.05 m from the initial ball position